src/code_review_agent/main.py

```diff
@@ -0,0 +1,247 @@
+    
+    return {
+        "status": "running",
+        "active_reviews": active_reviews,
+        "active_review_count": len(active_reviews),
+        "settings": {
+            "ai_model": settings.ai_model.value,
+            "max_files_to_review": settings.max_files_to_review,
+            "max_concurrent_reviews": settings.max_concurrent_reviews,
+            "environment": settings.environment.value
+        }
+    }
+
+
+@app.post("/webhook/github")
+async def github_webhook(request: Request):
+    """Handle GitHub webhook events."""
+    try:
+        # Get request body and signature
+        body = await request.body()
+        signature = request.headers.get("X-Hub-Signature-256")
+        
+        # Parse payload
+        payload_data = await request.json()
+        payload = GitHubWebhookPayload(**payload_data)
+        
+        # Handle webhook
+        result = await webhook_handler.handle_webhook(
+            payload=payload,
+            signature=signature,
+            payload_body=body
+        )
+        
+        # Update metrics
+        if result.get("status") == "started":
+            _metrics["total_reviews"] += 1
+        
+        return result
+        
+    except Exception as e:
+        logger.error(
+            "Error handling GitHub webhook",
+            error=str(e),
+            exc_info=True
+        )
+        _metrics["total_errors"] += 1
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Webhook processing failed: {str(e)}"
+        )
+
+
+@app.post("/review/manual")
+async def trigger_manual_review(request: ManualReviewRequest):
+    """Trigger a manual code review."""
+    try:
+        result = await webhook_handler.trigger_manual_review(
+            repo_full_name=request.repository_full_name,
+            pr_number=request.pull_request_number,
+            force=request.force_review
+        )
+        
+        # Update metrics
+        if result.get("status") == "started":
+            _metrics["total_reviews"] += 1
+        
+        return result
+        
+    except Exception as e:
+        logger.error(
+            "Error triggering manual review",
+            repo=request.repository_full_name,
+            pr_number=request.pull_request_number,
+            error=str(e),
+            exc_info=True
+        )
+        _metrics["total_errors"] += 1
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Manual review failed: {str(e)}"
+        )
+
+
+@app.get("/reviews/active")
+async def get_active_reviews():
+    """Get list of currently active reviews."""
+    return webhook_handler.get_active_reviews()
+
+
+@app.delete("/reviews/{request_id}")
+async def cancel_review(request_id: str):
+    """Cancel an active review."""
+    try:
+        cancelled = await webhook_handler.review_engine.cancel_review(request_id)
```

Comment:
This cancels via `review_engine`, but active review tasks are tracked in `GitHubWebhookHandler._active_tasks`. As written, cancellation won’t affect the background task started in the webhook handler. Consider exposing a public `cancel_review(request_id)` on the handler and using it here.

```suggestion
cancelled = await webhook_handler.cancel_review(request_id)
``` 

And in `GitHubWebhookHandler`:
```python
async def cancel_review(self, request_id: str) -> bool:
    task = self._active_tasks.pop(request_id, None)
    if task and not task.done():
        task.cancel()
        logger.info("Cancelled review", request_id=request_id)
        return True
    return False
```